plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/SQLServerSchema.java

```diff
@@ -272,7 +272,20 @@ public List<SQLServerObject> getChildren(@NotNull DBRProgressMonitor monitor) th
 
     @Override
     public SQLServerTableBase getChild(@NotNull DBRProgressMonitor monitor, @NotNull String childName) throws DBException {
-        return tableCache.getObject(monitor, this, childName);
+        SQLServerTableBase object = tableCache.getObject(monitor, this, childName);
+        if (object != null) {
+            return object;
+        }
+        // tempdb tables have special naming convention. See SQLServerUtils.stripTempdbTableName
+        if (database.getName().equalsIgnoreCase(SQLServerConstants.TEMPDB_DATABASE)) {
+            for (SQLServerTableBase table : tableCache.getAllObjects(monitor, this)) {
+                String name = SQLServerUtils.stripTempdbTableName(table.getName());
```

Comment:
This is a super-slow way to find a table. Again we can cache stripped table names in SQLServerTable.
Or even create class `SQLServerTableTemp extends SQLServerTableBase`.